### DISCUSSION TITLE
Bootloader Lock in Fajita/Enchillada

### DISCUSSION THREAD

**Discussion by EliasTheGrandMasterOfMistakes** [Divested-Mobile/DivestOS-Build#320] [Q&A] [answered]
Looks bootloader can be locked in Fajita and Enchilada, when i lock this bootloader i can recover if some problems occurs with EDL mode? 

If i want to build lineage for myself and locking bootloader, i can direct Locking using AVB custom keys and try to cherry-picking how divestOS is signed? exist a tutorial helping with building this task to use as reference?

**Comment by SkewedZeppelin** [comment, marked as answer]
Yep, OnePlus 5 and 6 are basically impossible to brick since EDL is available.
Just have all the recovery files available before hand ;)

As for signing you need a bunch of stuff, see:
prep
-  https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Generate_Signing_Keys.sh
- https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Generate_Verified_Boot_Hashes.sh
- https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Generate_Supporting_Files.sh

pre-build
- https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Common/Copy_Keys.sh
- https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Common/Enable_Verity.sh
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Patch.sh#L91
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Patch.sh#L444
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Patch.sh#L448
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Patch.sh#L453-L454

start build
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Functions.sh#L36
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/LineageOS-20.0/Functions.sh#L88

after build
- https://github.com/Divested-Mobile/DivestOS-Build/blob/f0de4dbf4c99d8874fbb8c7ee8173bb22f927f89/Scripts/Common/Functions.sh#L197-L485

**Reply by EliasTheGrandMasterOfMistakes** [reply]
Thank you for the helping, i will try to make a local build in future.
Sorry just a question, community had been banned in reddit, have some chance to be back?

**Reply by SkewedZeppelin** [reply]
I messaged Reddit admins but they say they are backlogged.

**Reply by EliasTheGrandMasterOfMistakes** [reply]
thank you i hope that community be backing.
in case of lineage21 these scripts can be used without bugs? and it's used after build or integrated?

**Reply by EliasTheGrandMasterOfMistakes** [reply]
> thank you i hope that community be backing. and it's used after build or integrated?

**Reply by SkewedZeppelin** [reply]
some of them are changes to the build, some of them are things that are generated before the build, and then some are used after the build
